haystack/components/generators/chat/hugging_face_local.py

```diff
@@ -11,7 +11,7 @@
 from typing import Any, Callable, Dict, List, Literal, Optional, Union, cast
 
 from haystack import component, default_from_dict, default_to_dict, logging
-from haystack.dataclasses import ChatMessage, ComponentInfo, StreamingCallbackT, ToolCall
+from haystack.dataclasses import AsyncStreamingCallbackT, ChatMessage, ComponentInfo, StreamingCallbackT, ToolCall
```

Comment:
as a heads up I did a refactor of this component in https://github.com/deepset-ai/haystack/pull/9455 so I'd appreciate if we could leave out the changes here and I can add them into mine once this PR is merged

Reply:
Hmm actually fine to leave. I'll handle the merge conflict rather after this PR is merged